lib/schema-builder/metadata/enum.metadata.ts

```diff
@@ -1,5 +1,16 @@
-export interface EnumMetadata {
-  ref: object;
+export interface EnumMetadataValuesMapOptions {
+  deprecationReason?: string;
+  description?: string;
+}
+
+export type EnumMetadataValuesMap<T extends object> = Record<
+  keyof T,
+  EnumMetadataValuesMapOptions
+>;
```

Comment:
```suggestion
export type EnumMetadataValuesMap<T extends object> = Partial<Record<
  keyof T,
  EnumMetadataValuesMapOptions
>>;
```
This should fix the test. I believe record types by default have the keys set as required.

Reply:
Thanks!